### PR TITLE
Coerce mount_point only when it is not root directory

### PR DESCRIPTION
**Pull request by jiokmiso**
Signed-off-by: JiSoo Kim <jskim910118@gmail.com>

## Description

Since #10614, mount resource is updated to have coerce in mount_point attribute. root directory
There were several issues around for root of the network mount, but the same change is also affecting the plain root directory. After the coercion, it is simply going to be nil. This PR is trying to address the issue by not applying the coercion when mount_point is either root directory or root of network mount.

## Related Issue

https://github.com/chef/chef/issues/12021

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/resource/mount.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Mount < Chef::Resource
         sensitive: true
 
       property :mount_point, String, name_property: true,
-               coerce: proc { |arg| arg.chomp("/") }, # Removed "/" from the end of str, because it was causing idempotency issue.
+               coerce: proc { |arg| (arg == "/" || arg.match?(":/$"))? arg : arg.chomp("/") }, # Removed "/" from the end of str, because it was causing idempotency issue.
```
Comment (lamont-granquist) [inline lib/chef/resource/mount.rb:45]: maybe `proc { |arg| arg.count("/") > 1 && arg.chomp("/") }` ?

Comment (jiokmiso) [reply lib/chef/resource/mount.rb:45]: I stayed with this implementation for the consistency across mount resource / provider, per https://github.com/chef/chef/blob/main/lib/chef/provider/mount.rb#L180 and https://github.com/chef/chef/blob/main/lib/chef/provider/mount/mount.rb#L206.
Do you want me to update the logic in these places too?

**Comment by ramereth** [comment]
This isn't going to revert the behavior in #11261 is it? I feel like we keep flapping back and forth on how this is implemented.

File: lib/chef/resource/mount.rb (hunk 1 above)
Comment (lamont-granquist) [reply lib/chef/resource/mount.rb:45]: okay i think after talking this out in issue triage that i'm buying the argument that the code as its written in these three places has been battle tested and does not have edge conditions that have been experienced while theoretically my suggestion could have edge conditions, so this code is fine as it is.

**Review by lamont-granquist — APPROVED** [review]

**Comment by tas50** [comment]
Thanks @jiokmiso. This will ship in the upcoming 17.5 release.

**Comment by jiokmiso** [comment]
@tas50 Is there a way to get this shipped in chef 16 also? That's where I am needing this fix mostly.

**Comment by tas50** [comment]
Next time we ship 16 we'll take a look at all the bug fixes and this should be part of that

**Comment by jiokmiso** [comment]
Thank you! I'll keep an eye on that issue.

**Pull request merged**